### PR TITLE
UI: Fix the resize cursor with respecting the item transformation

### PR DESCRIPTION
**Pull request by BGM99**
### Description
When rotating or flipping a scene item the resize cursor is not updated correctly. With this fix the cursor gets set according to the item rotation, the horizontal scale and the vertical scale. It handles all 90° and 45° rotations. All rotations in between are rounded up or down.

Before:
doubledip

After:

after1
after2
after3
after4

### Motivation and Context

fixes [#7247](https://github.com/obsproject/obs-studio/issues/7247)

### How Has This Been Tested?

I've tested it with nearly every rotation and transformation that is possible on windows with dev version.

What else should I test?

### Types of changes
 - Bug fix (non-breaking change which fixes an issue)

### Checklist:

- [x] My code has been run through [clang-format](https://github.com/obsproject/obs-studio/blob/master/.clang-format).
- [x] I have read the [**contributing** document](https://github.com/obsproject/obs-studio/blob/master/CONTRIBUTING.rst).
- [x] My code is not on the master branch.
- [x] The code has been tested.
- [x] All commit messages are properly formatted and commits squashed where appropriate.
- [x] I have included updates to all appropriate documentation.

### CODE REVIEW

**Comment by RytoEX** [comment]
If this fixes a specific GitHub Issue, please use the keywords "Fixes #\<Issue Number>". See:
* https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue
* https://docs.github.com/en/get-started/writing-on-github/working-with-advanced-formatting/using-keywords-in-issues-and-pull-requests

Please provide both before and after screenshots of the behavior.

**Comment by RytoEX** [comment]
How does this handle non-90° rotations? I suppose an equally valid question is how do we currently handle non-90° rotations?

**Comment by BGM99** [comment]
> How does this handle non-90° rotations? I suppose an equally valid question is how do we currently handle non-90° rotations?

There is no need to change anything on the non-90° rotations. When a item is rotated by 180°, then Top Left will be at the former position of Bottom Right. If you look at [window-basic-preview.cpp L.659 and L.660](https://github.com/obsproject/obs-studio/blob/master/UI/window-basic-preview.cpp#L659), you can see that both handles use the same cursor. The same applies to horizontal and vertical cursors. But with a 90° clockwise rotation there is the problem that Top Left will be displayed as Bottom Left, what causes the bug.

**Comment by RytoEX** [comment]
> > How does this handle non-90° rotations? I suppose an equally valid question is how do we currently handle non-90° rotations?
> 
> There is no need to change anything on the non-90° rotations. When a item is rotated by 180°, then Top Left will be at the former position of Bottom Right. If you look at [window-basic-preview.cpp L.659 and L.660](https://github.com/obsproject/obs-studio/blob/master/UI/window-basic-preview.cpp#L659), you can see that both handles use the same cursor. The same applies to horizontal and vertical cursors. But with a 90° clockwise rotation there is the problem that Top Left will be displayed as Bottom Left, what causes the bug.

To be clear, I'm not asking for changes. I'm just asking, "how does this handle non-90° rotations?" Put another way, could you please describe and/or provide screenshots of the before and after for that scenario? For example, 45° rotations.

**Comment by BGM99** [comment]
As I already mentioned, just rotations that are in between 45° and 135° or 225° and 315° are inverted.

So 45° remains the same:
45grad

The next possible rotation is around 50°, from here my fix inverts the cursor:
50grad

70°, as we get closer to 90°, my fix makes more sense:
70grad

125° with another handle:
125grad


I hope it is now clearer how it currently works and what the scope of my change is.

File: UI/window-basic-preview.cpp [hunk 1]
```diff
@@ -658,9 +658,48 @@ void OBSBasicPreview::UpdateCursor(uint32_t &flags)
...
-	if (cursor().shape() != Qt::ArrowCursor)
+	if ((cursor().shape() != Qt::ArrowCursor) || flags == 0)
 		return;
 
+	if (flags & ITEM_ROT) {
+		setCursor(Qt::OpenHandCursor);
+		return;
+	}
+
+	float rot = obs_sceneitem_get_rot(stretchItem);
```
Comment (PatTheMav) [inline UI/window-basic-preview.cpp:669]: Rename `rot` to `rotation`, `q` to `quadrant`, and `corner` to `isCorner`. We have the storage, we can write out variable names. 😄

**Review by PatTheMav — COMMENTED** [review]
Pretty nifty amount of bit-shifting but this all seems correct - Opposing diagonal pairs are treated as equivalent to simplify calculations and at some more extreme angles (or right at the edge of quadrants) the chosen handle might not be perfect, but it's still in the overall correct direction (compared to the current status quo).

**Comment by BGM99** [comment]
@PatTheMav Thx for the review. Since I use a circular sector of 45°, the correct term would be "octant". By "q" I actually meant quotient, but octant is more precise. It is true that opposing diagonal pairs are treated as equal, but it shouldn't be a problem since it's irrelevant for the UI and the flag won't be used afterwards. But its important to mention that the flag is incorrect afterwards (when there is a rotation), regardless of this. If this is a problem somehow, then we should call UpdateCursor() with call by value.

**Comment by PatTheMav** [comment]
"Octant" would be actually correct, yeah.

I agree that the interchangeable use of the handles is fine, insofar as it doesn't have a _visual_ impact (the function is entirely unaffected by this).

Fixing that would require a bit more complex bit arithmetic (e.g. just shifting the left/right part to the top/bottom part for non-perpendicular angles wouldn't suffice).

**Comment by BGM99** [comment]
I have tested it visually for all 8 rotation cases, and the resize functionality still works fine. So is this approach mergeable?

**Review by PatTheMav — APPROVED** [review]

**Comment by PatTheMav** [comment]
@BGM99 could you please change the commit title to something like `UI: Fix resize cursor to respect item transformation`.

**Review by RytoEX — APPROVED** [review]
Behavior with this PR is certainly better than it is now. Seems fine.

**Pull request merged**